### PR TITLE
Terminate thread on exit

### PR DESCRIPTION
**Pull request by breerly**
Looks like this works.

@blampe @abhinav

### CODE REVIEW

File: threadloop/threadloop.py [hunk 1]
```diff
@@ -26,7 +27,14 @@ def __exit__(self, *args):
...
         self.thread = Thread(target=self.io_loop.start)
-        self.thread.start()
+
+        # terminate thread & cleanup
+        # on system exit or keyboard interrupt
+        try:
+            self.thread.start()
+        except (KeyboardInterrupt, SystemExit):
+            self.stop()
+            sys.exit()
```
Comment (blampe) [inline threadloop/threadloop.py]: `raise`

Comment (breerly) [inline threadloop/threadloop.py]: @blampe i was wrong - this doesn't work :\

**Comment by blampe** [comment]
BOOM headshot

**Comment by abhinav** [comment]
:disappointed:

**Comment by breerly** [comment]
We're going to go with this for now - it's the safest approach and a catch-all.

**Comment by breerly** [comment]
;( ;( ;(

**Pull request merged**